src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java

```diff
@@ -0,0 +1,104 @@
+    private static final int PERFORMER_CODE_INDEX_IN_EDIFACT_STRING = 3;
+
+    @NonNull
+    private final String identifier;
+    private final HealthcareRegistrationIdentificationCode code;
+    @NonNull
+    private final String performingOrganisationName;
+    @NonNull
+    private final String partyName;
+
+    public static PerformerNameAndAddress fromString(String edifactString) {
+        if (!edifactString.startsWith(KEY_QUALIFIER)) {
+            throw new IllegalArgumentException(
+                "Can't create " + PerformerNameAndAddress.class.getSimpleName() + " from " + edifactString
+            );
+        }
+
+        String[] keySplit = Split.byPlus(edifactString);
+        String[] colonSplit = Split.byColon(keySplit[2]);
+        String performerID = colonSplit[0];
+
+        if (performerID.isBlank()) {
+            // if identifier is blank - organisation/department
+            String performingOrganisationName = keySplit[PERFORMING_NAME_INDEX_IN_EDIFACT_STRING];
+            return new PerformerNameAndAddress("", null, performingOrganisationName, "");
+        } else {
+            String performerCode = colonSplit[1];
+            String performerPartyName = keySplit[PERFORMING_NAME_INDEX_IN_EDIFACT_STRING];
+            System.out.println(performerPartyName);
+            return new PerformerNameAndAddress(
+                performerID,
+                HealthcareRegistrationIdentificationCode.fromCode(performerCode),
+                "",
+                performerPartyName
+            );
+        }
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        if (this.identifier.isBlank()) {
+            return QUALIFIER + "+++" + performingOrganisationName;
+        } else {
+            return QUALIFIER + PLUS_SEPARATOR + this.identifier + COLON_SEPARATOR
+                + this.code.getCode() + PLUS_SEPARATOR + PLUS_SEPARATOR + this.partyName;
+        }
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (this.identifier.isBlank()) {
+            if (performingOrganisationName.isBlank()) {
+                throw new EdifactValidationException(getKey() + ": Attribute performingOrganisationName is required");
+            }
+        } else {
+            if (code.getCode().isBlank()) {
+                throw new EdifactValidationException(getKey() + ": Attribute code is required");
+            }
+            if (partyName.isBlank()) {
+                throw new EdifactValidationException(getKey() + ": Attribute partyName is required");
+            }
+        }
+    }
+
+    public String toEdifact() throws EdifactValidationException {
+        this.validate();
+        return this.getKey() + PLUS_SEPARATOR + this.getValue() + TERMINATOR;
+    }
```

Comment:
Is this used?

Reply:
We currently use it in several tests in `PerformerNameAndAddress.java` but I'm not sure where we would use it in our application

Reply:
Initially we were using the superclass (`Segment.java`) version of the method, but once I added the variation of the segment (the Practitioner part where `identifier` was used, this caused an issue because `identifier` and `code` had a colon separator instead of a plus separator.

Reply:
Okay, so as an overriding implementation it should have `@Override`, but this is overriding with the exact same implementation as that in `Segment`, so it should be unnecessary.

(Running all unit tests without it passes, anyway.)

Reply:
Added `Override` annotation in 1bbb683